xbmc/addons/kodi-dev-kit/include/kodi/addon-instance/Game.h

```diff
@@ -1179,6 +1308,71 @@ class ATTR_DLL_LOCAL CInstanceGame : public IAddonInstance
         ->SetCheat(index, enabled, code);
   }
 
+  inline static GAME_ERROR ADDON_RCGenerateHashFromFile(const AddonInstance_Game* instance,
+                                                        char* hash, 
+                                                        int consoleID,
+                                                        const char* filePath)
+  {
+    return static_cast<CInstanceGame*>(instance->toAddon->addonInstance)
+        ->RCGenerateHashFromFile(hash, consoleID, filePath);
+  }
+
+  inline static GAME_ERROR ADDON_RCGetGameIDUrl(const AddonInstance_Game* instance,
+                                                char* url,
+                                                size_t size,
+                                                const char* hash)
+  {
+    return static_cast<CInstanceGame*>(instance->toAddon->addonInstance)
+        ->RCGetGameIDUrl(url, size, hash);
+  }
+
+  inline static GAME_ERROR ADDON_RCGetPatchFileUrl(const AddonInstance_Game* instance,
+                                                   char* url,
+                                                   size_t size,
+                                                   const char* username,
+                                                   const char* token,
+                                                   unsigned gameID)
+  {
+    return static_cast<CInstanceGame*>(instance->toAddon->addonInstance)
+        ->RCGetPatchFileUrl(url, size, username, token, gameID);
+  }
+
+  inline static GAME_ERROR ADDON_RCPostRichPresenceUrl(const AddonInstance_Game* instance,
+                                                       char* url,
+                                                       size_t urlSize,
+                                                       char* postData,
+                                                       size_t postSize,
+                                                       const char* username,
+                                                       const char* token,
+                                                       unsigned gameID,
+                                                       const char* richPresence)
+  {
+    return static_cast<CInstanceGame*>(instance->toAddon->addonInstance)
+        ->RCPostRichPresenceUrl(url, urlSize, postData, postSize, username, token, gameID,
+                                richPresence);
+  }
```

Comment:
To become as C++ above, it can be as:
```cpp
  inline static GAME_ERROR ADDON_RCPostRichPresenceUrl(const AddonInstance_Game* instance,
                                                       char** url,
                                                       char** postData,
                                                       const char* username,
                                                       const char* token,
                                                       unsigned gameID,
                                                       const char* richPresence)
  {
    std::string cppUrl;
    std::string cppPostData;
    GAME_ERROR ret = static_cast<CInstanceGame*>(instance->toAddon->addonInstance)
        ->RCPostRichPresenceUrl(cppUrl, cppPostData, username, token, gameID, richPresence);
    if (!cppUrl.empty())
      *url = strdup(cppUrl.c_str());
    if (!cppPostData.empty())
      *postData = strdup(cppPostData.c_str());

    return ret;
  }
```